File: tlb/05/solutions/head0.py
#!/usr/bin/python

import re, sys, itertools

n_lines = 10

if len(sys.argv) > 1 and re.match(r'-[0-9]+', sys.argv[1]):
    n_lines = int(sys.argv.pop(1)[1:])

if len(sys.argv) == 1:
    sys.argv.append("-")

for filename in sys.argv[1:]:
    try:
        print("==> %s <==" % filename)
        if filename == "-":
            stream = sys.stdin
        else:
            stream = open(filename)
        for line in itertools.islice(stream, n_lines):
            sys.stdout.write(line)
        if filename != "-":
            stream.close()
    except IOError as  e:
        (errno, strerror) = e.args
        print("%s: can not open: %s" % (filename, strerror))

File: tlb/05/solutions/head0.pl
#!/usr/bin/perl -w
if ($ARGV[0] =~ /-[0-9]+/) {
    $max = shift @ARGV;
    $max =~ s/-//;
} else {
    $max = 10;
}
# default is stdin if no files specified
$ARGV[0] = "-" if @ARGV == 0;
foreach $file (@ARGV) {
    open my $input, '<', $file or die "$file: can not open: $!\n";
    print "==> $file <==\n";
    @lines = <$input>;
    print @lines[0..$max-1];
    close $input;
}